units/host_status.go

```diff
@@ -180,3 +185,51 @@ func setCloudHostStatus(ctx context.Context, m cloud.Manager, h host.Host, hostS
 	})
 	return nil
 }
+
+func (j *cloudHostReadyJob) setNextState(h host.Host) error {
+	switch h.Distro.BootstrapSettings.Method {
+	case distro.BootstrapMethodUserData:
+		// Task hosts are not provisioned so we can skip the provisioning state.
+		return errors.Wrapf(h.SetProvisionedNotRunning(), "error marking host %s as provisioned not running", h.Id)
+	case distro.BootstrapMethodNone:
+		// We're done provisioning user data hosts. The user data script will do the rest.
```

Comment:
This comment and the one above it on the `distro.BootstrapMethodUserData` case should be swapped. Also, can we call them "host.create hosts" or "hosts created by tasks" instead of task hosts?